# ML-02 — Research Question and Provisional Lane

[![Open In Colab](https://colab.research.google.com/assets/colab-badge.svg)](https://colab.research.google.com/github/moustafaban2005-dot/flyrank-ml-internship-moustafa/blob/main/work/notebooks/w01_research_question.ipynb?flush_cache=true)

This skeleton is yours to fill. Work the sections **in order** — each one has a one-line hint. Simple words, honest numbers.

> Working with an AI assistant? Tell it to read `skills/README.md` first and load the one skill this assignment names on its card.

## 1. My lane (or freestyle) and why

*Name your lane — or say 'freestyle' and describe your own question. One short paragraph: why this one?*

Lane: Lane 2 — Refresh / Content Opportunity Scoring. I'm picking this because the first two notebooks already got me comparing a hand-written rule against a learned model to flag pages worth reviewing first, and I ran into a real leakage problem while doing it (trend_pct predicting itself). This lane extends that same skill — building a ranked queue of pages to act on — instead of starting a totally new kind of analysis from scratch.

In [10]:
#no code needed for this section

## 2. The question: decision, action, cost of a wrong call

*What decision does your work improve? Who acts on it? What does a wrong recommendation cost?*

The question: Which existing pages should be reviewed first for refresh, expansion, or pruning?
Unit of analysis: a single page.
The decision: which pages a content editor tackles first this week, out of potentially thousands of pages — since nobody has time to review everything.
The action: an editor gets a ranked list of pages with a suggested action per page (refresh, expand, prune, monitor) and a reason code explaining why it was flagged (e.g. "declining traffic with steady demand," "stale but still getting impressions").
The cost of a wrong call: two ways this can go wrong. False positive — an editor spends hours refreshing a page that wouldn't have improved anyway, wasting time. False negative — a page that's quietly losing traffic gets ignored and keeps declining until it's much harder to recover. Since editor time is limited, the real cost isn't just "wrong page," it's "right page not reviewed because the wrong one got picked instead."
Why data/ML helps at all: with a small number of pages, a person could just eyeball them. At the scale FlyRank operates on (tens of thousands of pages across clients), nobody can manually check each one — a model that ranks pages by "worth checking first" turns an impossible manual task into a short, reviewable list.

In [11]:
#no code needed for this section

## 3. Quick look at the data (2-3 real numbers)

*Load the starter CSV below and show 2-3 real numbers that make your lane look worth the next 7 weeks.*

Loading the starter dataset to check whether "declining pages" and "stale pages" actually exist in meaningful numbers — if barely any pages fit these patterns, this lane isn't worth 7 weeks.

In [12]:
!git clone https://github.com/moustafaban2005-dot/flyrank-ml-internship-moustafa.git
%cd flyrank-ml-internship-moustafa

Cloning into 'flyrank-ml-internship-moustafa'...
remote: Enumerating objects: 119, done.
remote: Counting objects: 100% (119/119), done.
remote: Compressing objects: 100% (76/76), done.
remote: Total 119 (delta 35), reused 94 (delta 27), pack-reused 0 (from 0)
Receiving objects: 100% (119/119), 1.84 MiB | 7.71 MiB/s, done.
Resolving deltas: 100% (35/35), done.
/content/flyrank-ml-internship-moustafa/flyrank-ml-internship-moustafa


In [13]:
import pandas as pd

df = pd.read_csv("data/raw/content_refresh_anonymized.csv")

print("Total pages:", len(df))
print("Pages with declining trend:", (df["trend_direction"] == "down").sum())
print("Pages with 0 updates in 90+ days (if column exists):")
if "days_since_last_update" in df.columns:
    print((df["days_since_last_update"] > 90).sum())

Total pages: 30000
Pages with declining trend: 16262
Pages with 0 updates in 90+ days (if column exists):
9345


Observed: out of 30,000 total pages, 16,262 (about 54%) show a declining trend, and 9,345 (about 31%) haven't been updated in 90+ days. More than half the inventory is trending down, and roughly a third is stale — that's a meaningful chunk of pages, way too many for anyone to review by hand. This confirms there's a real prioritization problem worth building a ranked queue for over the next 7 weeks.

## 4. Careful words: what I can and can't claim

*Write what your work will be able to say (observed, directional, decision-support) — and what it never will (causal proof, 'predicting Google').*

What I can say: which pages look worth reviewing first, based on observed patterns in traffic, position, and freshness — directional, decision-support signals to help an editor prioritize.
What I can never say: that a page is declining because of a specific cause, that refreshing it will definitely fix the ranking, or anything about how Google's algorithm actually works. This is pattern-matching on anonymized data, not a controlled experiment — correlation, not causation.

## Self-check

Before you submit, confirm each line honestly:

- [ ] Every section above is filled — markdown thinking AND the code that backs it
- [ ] The notebook runs top to bottom with no errors (Runtime → Run all)
- [ ] No client names, URLs, or private queries anywhere
- [ ] My claims use careful words: observed, measured, directional, decision-support
- [ ] Committed to my repo under `work/notebooks/` — then submit your repo URL on the card. Done.